src/hotspot/share/opto/compile.cpp

```diff
@@ -4514,7 +4514,9 @@ bool Compile::coarsened_locks_consistent() {
     bool modified = false; // track locks kind modifications
     Lock_List* locks_list = (Lock_List*)_coarsened_locks.at(i);
     uint size = locks_list->size();
-    if (size != locks_list->origin_cnt()) {
+    if (size == 0) {
+      unbalanced = false; // All locks were eliminated - good
+    } else if (size != locks_list->origin_cnt()) {
```

Comment:
Makes sense.

Reply:
Yes. For example, if one of branches in case 1 is eliminated after locks/unlocks marked as coarsened:
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/callnode.cpp#L1779

jbb2005 does not have such cases. And It would be really hard to separate such case  from failing case in current implementation.

I have RFE [8268571](https://bugs.openjdk.java.net/browse/JDK-8268571) it improve locks elimination I hope I can work on in JDK 18.